201-vm-linux-dynamic-data-disks/diskSelection.json

```diff
@@ -59,261 +59,182 @@
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "5": [
-        {
-          "name": "datadisk4",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 4,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk4.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "6": [
-        {
-          "name": "datadisk5",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 5,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk5.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "7": [
-        {
-          "name": "datadisk6",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 6,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk6.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "8": [
-        {
-          "name": "datadisk7",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 7,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk7.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "9": [
-        {
-          "name": "datadisk8",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 8,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk8.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
-        }
-      ],
-      "10": [
-        {
-          "name": "datadisk9",
-          "diskSizeGB": "[parameters('diskSizeGB')]",
-          "lun": 9,
-          "vhd": {
-            "Uri": "[concat('http://',parameters('diskStorageAccountName'),'.blob.core.windows.net/',parameters('vmStAccountContainerName'),'/',parameters('masterName'),'-datadisk9.vhd')]"
-          },
-          "caching": "None",
-          "createOption": "Empty"
+    "uriPrefix": "[uri(parameters('diskStorageAccountUri'), concat(parameters('vmStAccountContainerName'), '/', parameters('masterName')))]",
+    "dataDisks": [
```

Comment:
I think I got the idea, you probably mean [this](https://docs.microsoft.com/en-us/azure/architecture/building-blocks/extending-templates/objects-as-parameters) approach.
I think we could try to use it.